Comp-Bot/src/main/java/frc/robot/lib/drivers/TalonSRX.java

```diff
@@ -71,4 +71,4 @@ public static WPI_TalonSRX createTalonSRX(int deviceId, int leaderId) {
     mLogger.info("Created follower TalonSRX {}]", deviceId);
     return talon;
   }
-}
\ No newline at end of file
+}
```

Comment:
Proposed changes:

//New method: meant so the chunk of code does not need to be written in both createLeader and createFollower
 public static void setSensorConfig(WPI_TalonSRX talon) {
    final ErrorCode sensorPresent = talon.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 100);
    if (sensorPresent != ErrorCode.OK) {
      mLogger.error("Could not detect encoder due EC: [{}]", sensorPresent);
    }
    talon.setSensorPhase(true);
    talon.configVelocityMeasurementPeriod(VelocityMeasPeriod.Period_50Ms, RobotMap.kLongCANTimeoutMs);
    talon.configVelocityMeasurementWindow(1, RobotMap.kLongCANTimeoutMs);
    talon.configClosedloopRamp(0.0, RobotMap.kLongCANTimeoutMs);
  }

// Changed method: Instead of configuring the sensor here, is moved to method
  public static void createLeader(WPI_TalonSRX talon, boolean hasSensor) {
    setDefaultConfig(talon);
    setSensorConfig(talon);
    talon.setStatusFramePeriod(StatusFrameEnhanced.Status_2_Feedback0, 20, RobotMap.kLongCANTimeoutMs);
    talon.set(ControlMode.PercentOutput, 0.0);
  }

// New Method: Overloaded createfollower, for when a follower has a sensor
  public static void createFollower(WPI_TalonSRX talon, int leaderId, boolean hasSensor) {
    setDefaultConfig(talon);
    setSensorConfig(talon);
    talon.setStatusFramePeriod(StatusFrameEnhanced.Status_2_Feedback0, 160, RobotMap.kLongCANTimeoutMs);
    talon.set(ControlMode.Follower, leaderId);  
  }

// New Method: Overloaded createTalonSRX for when a follower has a sensor
 public static WPI_TalonSRX createTalonSRX(int deviceId, int leaderId, boolean hasSensor) {
    final WPI_TalonSRX talon = new WPI_TalonSRX(deviceId);
    createFollower(talon, leaderId, hasSensor);
    mLogger.info("Created follower TalonSRX [{}] with an encoder", deviceId);
    return talon;
  }
}

Reply:
I reworked the whole code...including the Junit tests